### PR TITLE
Adding Opensearch field endpoint to support AWS Opensearch

### PR DESCRIPTION
**Pull request by jwitrick**
This PR addresses the missing ability to use the logging-operator with a secure AWS Opensearch cluster.

Solves this issue: https://github.com/kube-logging/logging-operator/issues/1477

### CODE REVIEW

File: pkg/sdk/logging/model/output/opensearch.go [hunk 1]
```diff
@@ -278,3 +294,34 @@ func (e *OpenSearchOutput) ToDirective(secretLoader secret.SecretLoader, id stri
...
 }
+
+
+
+type OpenSearchEndpointCredentials struct {
+	// AWS region. It should be in form like us-east-1, us-west-2. Default nil, which means try to find from environment variable AWS_REGION.
+	Region string `json:"region,omitempty"`
+
+	// AWS connection url.
+	Url string `json:"url,omitempty"`
```
Comment (pepov) [inline pkg/sdk/logging/model/output/opensearch.go:305]: nit: shouldn't this be mandatory?

**Review by pepov — APPROVED** [review]
looks wonderful, thanks!

**Review by OverOrion — COMMENTED** [review]
Thank you for your contribution @jwitrick! Currently the [CI fails ](https://github.com/kube-logging/logging-operator/actions/runs/6894811966/job/18775701005?pr=1582)because some generated code has not been committed.
Could you run `make check-diff` and commit the newly generated files?

File: pkg/sdk/logging/model/output/opensearch.go (hunk 1 above)
Comment (jwitrick) [reply pkg/sdk/logging/model/output/opensearch.go:305]: Looking at the plugin i think you are correct it should be. Fixing now.

Comment (jwitrick) [reply pkg/sdk/logging/model/output/opensearch.go]: Changes pushed. Please take another look.

**Pull request merged**